### DISCUSSION TITLE
Grid values assigned, unexpected result

### DISCUSSION THREAD

**Discussion by alessandroAM** [AcademySoftwareFoundation/openvdb#1323] [Q&A]
Hello, after studying some examples I made a FloatGrid and populated it with a vertical gradient, so that voxel values range from 0 to 1.
Result is in this image
![sshot-1](https://user-images.githubusercontent.com/24398090/156857822-c2ae82f2-6a9e-4a35-857a-1a302e014f29.jpg)

Granted that setting voxels at 0 does indeed produce a 0 density output, It seems to me that the scale of values is off because the density basically reach high values immediately.

Attached an image showing densities sampled at different values: 
![values](https://user-images.githubusercontent.com/24398090/156857914-4a25f348-b107-4b67-8dd8-fcf7b1396d19.jpg)

So obviously I'm doing something wrong, maybe there is a setting for min and max value for the grid?

Here is a part of the code I'm using, related to assigning values to the grid.

```
   for (x = bmin.x(); x <= bmax.x(); x+=inc)   // grid range is -0.5 to 0.5
    {        
        for (y = bmin.y(); y <= bmax.y(); y+= inc)
        {
            for (z = bmin.z(); z <= bmax.z(); z+= inc)
            {
                noiseValue =  y / (bmax.y() - bmin.y()); // values from 0 to 1
                dense.setValue(xyz, noiseValue);
            }
        }
    }

    openvdb::FloatGrid::Ptr vdbGrid = openvdb::FloatGrid::create(0.0f);   
    openvdb::tools::copyFromDense(dense, *vdbGrid, 0.01);
    vdbGrid->setName("density");
    vdbGrid->setGridClass(openvdb::GRID_FOG_VOLUME);
    std::ofstream ofile2("test.vdb", std::ios_base::binary);
```

**Comment by jmlait** [comment]
Hello,

It is not clear what you are using to visualize the densities.  The code seems right - that will make a zero-one gradient, assuming xyz is set to a Coord vector of x,y,z.  But if you view a cube cross wise, the result is an integration of the density over the cube.  So one would expect a very swift alpha transition from black to white, depending on the thickness.  As soon as your cube is optically thick additional density stops having an effect.
